rfcs/text/0000-matrix-service.md

```diff
@@ -0,0 +1,418 @@
+| `/filters`            | `GET` | Returns a list of available filter values.                   |
+| `/filters/<filter>`   | `GET` | Returns more information about a specific filter.            |
+| `/fields`             | `GET` | Returns a list of available metadata field values.           |
+| `/fields/<field>`     | `GET` | Returns more information about a specific metadata field.    |
+| `/formats`            | `GET` | Returns a list of available matrix formats.                  |
+| `/formats/<format>`   | `GET` | Returns more information about a specific matrix format.     |
+| `/features`           | `GET` | Returns a list of available expression feature types.        |
+| `/features/<feature>` | `GET` | Returns more information about a specific expression feature type. |
+
+### POST /matrix
+
+To request a matrix, users must supply a metadata filter expression for which matching cells will be included in the generated matrix. Optionally, they may also specify a list of metadata fields to be exported per cell in the matrix, the output file format and the feature type to describe. Below is a sample request body and a description of the four fields:
+
+```
+{
+	'filter': dict,
+	'fields': list,
+	'format': string,
+	'feature': string
+}
+```
+
+#### Filter
+
+The filter object is inspired by the [Genomics Data Commons (GDC) API](https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#filters-specifying-the-query). To enable users the ability to express complex metadata queries, the object supports nested composition of base filter objects to create a tree of AND/OR statements. The API deviates from the GDC API specification by defining two types of base filter objects:
+
+*Table 1.3 Base filter object types*
+
+| Filter object     | Specification                                                |
+| ----------------- | ------------------------------------------------------------ |
+| Comparison Filter | {<br />   `op`: one of `[ =, !=, >, <, >=, <=, in ]`,<br />   `field`: a metadata field,<br />   `value`: string \| int \| list <br />} |
+| Logical Filter    | {<br />   `op`: one of `[ and, or, not ]`,<br />   `value`: array of >=2 filter objects `if op==and|or`, filter object `if op==not` <br />} |
+
+Comparison Filters specify a condition on a metadata field and Logical Filters implement `and`, `or` and `not` logic. Note that the `value` field of a Logical Filter accepts a base filter object and is the mechanism for nesting chains of AND/OR statements. For a Comparison Filter's `field` parameter, the list of accepted metadata filters is available at the `/filters` endpoint. More information about a filter such as a description and a set of valid values can be found at the `/filters/<filter>` endpoint. See [Auxiliary Endpoints](#auxiliary-endpoints) for more details on these.
+
+Ex. Select all full length cells:
+
+```
+"filter": {
+  "op": ">=",
```

Comment:
Is this meant to be "=" instead of ">="?

Reply:
Ah, yup! Good eye